### PR TITLE
Addressing Issue #1145 - Breadcrumb: Customize icon used

### LINKED ISSUE #1145: Breadcrumb: Customize icon used
Respected All,  

is it anyway to change the ms-Icon--ChevronRight with any other Icon, for example ms-Icon--ChevronLeft
in Breadcrumb? According to my understanding,  there is no option to change the icon.

Best Regards

Umer Javaid

### PR DESCRIPTION
**Pull request by oengusmacinog**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #1145 
- [x] Include a change request file using `$ npm run change`

#### Description of changes

Added dividerAs prop to Breadcrumb component allowing the user to pass a custom JSX element to be used as the trail divider icon.

### CODE REVIEW

**Comment by micahgodbolt** [comment]
@dzearing wanted to make sure that this change wouldn't cause problems with the approach you took in 6.0.

File: packages/office-ui-fabric-react/src/components/Breadcrumb/Breadcrumb.tsx [hunk 1]
```diff
@@ -65,7 +65,7 @@ export class Breadcrumb extends BaseComponent<IBreadcrumbProps, any> {
   }
 
   private _onRenderBreadcrumb = (data: IBreadCrumbData) => {
-    const { className, ariaLabel, onRenderItem = this._onRenderItem, overflowAriaLabel } = data.props;
+    const { className, ariaLabel, dividerAs, onRenderItem = this._onRenderItem, overflowAriaLabel } = data.props;
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Breadcrumb/Breadcrumb.tsx]: Hmm. Why not default dividerAs to Icon.

```ts
const { dividerAs = Icon } = props;
```

File: packages/office-ui-fabric-react/src/components/Breadcrumb/Breadcrumb.tsx [hunk 2]
```diff
@@ -99,7 +103,7 @@ export class Breadcrumb extends BaseComponent<IBreadcrumbProps, any> {
                     directionalHint: DirectionalHint.bottomLeftEdge
                   } }
                 />
-                { <Icon
+                { dividerAs || <Icon
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Breadcrumb/Breadcrumb.tsx]: i'm not sure this logic works.

File: packages/office-ui-fabric-react/src/components/Breadcrumb/Breadcrumb.types.ts [hunk 3]
```diff
@@ -25,6 +25,12 @@ export interface IBreadcrumbProps extends React.Props<Breadcrumb> {
    */
   className?: string;
 
+  /**
+   * Render a custom divider in place of the default chevron '>'
+   * as a custom JSX element
+   */
+  dividerAs?: JSX.Element;
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Breadcrumb/Breadcrumb.types.ts]: This should be `IComponentAs<IIconProps>` right?

Comment (oengusmacinog) [reply packages/office-ui-fabric-react/src/components/Breadcrumb/Breadcrumb.types.ts]: What if they want to use something other than a fabric Icon component as the divider 'icon'?

File: packages/office-ui-fabric-react/src/components/Breadcrumb/Breadcrumb.tsx (hunk 1 above)
Comment (oengusmacinog) [reply packages/office-ui-fabric-react/src/components/Breadcrumb/Breadcrumb.tsx]: I originally wrote it this way actually but it seemed messier and harder to follow to put it default icon in a constant like so:
```
<Icon
    className={ css('ms-Breadcrumb-chevron', styles.chevron) }
    iconName={ getRTL() ? 'ChevronLeft' : 'ChevronRight' }
/>
```
Or are you suggesting something else?

Comment (dzearing) [reply packages/office-ui-fabric-react/src/components/Breadcrumb/Breadcrumb.tsx]: I am suggesting this:

```
const { dividerAs: Divider = Icon } = props;

return ( 
 ...
  <Divider className={ ... } iconName={ ... } />
...
);
```

File: packages/office-ui-fabric-react/src/components/Breadcrumb/Breadcrumb.types.ts (hunk 3 above)
Comment (dzearing) [reply packages/office-ui-fabric-react/src/components/Breadcrumb/Breadcrumb.types.ts]: The props we pass in are important. But it doesn't mean that they can't use something else.

For example, your span case doesn't need the props:

```
const divider = () => <span>*</span>;

<Breadcrumb dividerAs={ divider } />
```
or
```
<Breadcrumb dividerAs="*" />
```
or
```
<Breadcrumb dividerAs={ iconProps => <div>{ iconProps.iconName }</div> } />
```

File: packages/office-ui-fabric-react/src/components/Breadcrumb/__snapshots__/Breadcrumb.test.tsx.snap [hunk 4]
```diff
@@ -136,16 +136,6 @@ exports[`Breadcrumb renders breadcumb correctly 1`] = `
                 TestText4
               </div>
             </span>
-            <i
-              aria-hidden={true}
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Breadcrumb/__snapshots__/Breadcrumb.test.tsx.snap]: This seems concerning. What is the icon missing in the snapshot?

Comment (dzearing) [reply packages/office-ui-fabric-react/src/components/Breadcrumb/__snapshots__/Breadcrumb.test.tsx.snap]: Also can you please add a separate simple snapshot test that renders the breadcrumb with a different icon?

**Pull request closed without merging**